app/src/main/java/com/example/QArmy/MainActivity.java

```diff
@@ -69,11 +113,66 @@ public void onPageSelected(int position) {
                 bottomNavigationView.getMenu().getItem(position).setChecked(true);
                 prevMenuItem = bottomNavigationView.getMenu().getItem(position);
             }
+
             @Override
             public void onPageScrollStateChanged(int state) {
 
             }
         });
     }
 
+    /**
+     * Create toolbar menu
+     * @param menu The toolbar menu
+     * @return
+     */
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.toolbar_menu, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    /**
+     * Perform action when toolbar button is pressed
+     * @param item The toolbar item selected
+     * @return
+     */
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        if (item.getItemId() == R.id.action_add) {
+            scanCode();
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    /**
+     * Open the camera to scan the QR code.
+     */
+    private void scanCode() {
```

Comment:
There's a linter-related error that's stopping things from building here (for the call on line 168). I took a look into it and it seems to be a problem with the linter (not your code), but to get around the linter, I'd add the @SupressLint("MissingPermission") annotation to the method so the linter ignores and we can still finish building the gradle and running the tests.